app/code/Magento/InventoryInStorePickup/Model/SearchResult/ExtractStrategy.php

```diff
@@ -0,0 +1,52 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryInStorePickup\Model\SearchResult;
+
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\InventoryApi\Api\Data\SourceSearchResultsInterface;
+use Magento\InventoryInStorePickupApi\Api\Data\SearchRequestInterface;
+use Magento\InventoryInStorePickupApi\Model\SearchResult\ExtractStrategyInterface;
+
+/**
+ * @inheritdoc
+ */
+class ExtractStrategy implements ExtractStrategyInterface
+{
+    /**
+     * @var DistanceBasedStrategy
+     */
+    private $distanceBasedStrategy;
+
+    /**
+     * @param DistanceBasedStrategy $distanceBasedStrategy
+     */
+    public function __construct(
+        DistanceBasedStrategy $distanceBasedStrategy
```

Comment:
Is it just a decorator for `\Magento\InventoryInStorePickup\Model\SearchResult\DistanceBasedStrategy`? Why cannot we use it directly and remote this class?

Reply:
I think something can be rethinked there.
For now we have: default extract and extract source, ordered by distance.
I think it is nice to have extension point there to have possibility to add new strategy for sources  extraction.

Reply:
Discussed offline.
Refactored.
Please check.